client/components/NavLink.jsx

```diff
@@ -1,18 +1,152 @@
+                headers: new Headers({
+                    "Content-Type": "application/json",
+                    Accept: "application/json"
+                })
+            })
+            .then(result => {
+                if (result.status === 200) {
+                    console.log("sign in successful");
+                    this.setState(prevState => ({
+                        modal: !prevState.modal,
+                        login: true,
+                        username: this.state.id
+                      }));
+                } else if (result.status === 400) {
+                    alert("Wrong Password!")
+                } else {
+                    alert("Username doesn't exist, Please Sign up!")
+                }
+            })
+            .catch(error => {
+                console.error(error, "postRequest error");
+            });
+        }
+        event.preventDefault();
+    }
+
+    signup() {
+        if (this.state.modalState === "Sign In") {
+            this.setState({modalState: 'Sign Up'});
+        } else {
+            var userInfo = {id: this.state.id, password: this.state.password};
+            return fetch('http://localhost:5000/signup', {
+                credentials: 'same-origin',
+                method: 'POST',
+                body: JSON.stringify(userInfo),
+                headers: new Headers({
+                    'Content-Type': 'application/json',
+                    'Accept': 'application/json'
+                }),
+            }).then((result) => {
+                if (result.status === 200) {
+                    alert("Signup successful!");
+                    this.setState({modalState: "Sign In"});
+                }
+                else {
+                    alert("Existing username please try other ID");
+                }
+            }).catch((error) => {
+                console.error(error, 'postRequest error');
+            })
+        }
+    }
+
     render() { 
         return ( 
             <nav className="navbar">
                 <Link to="/">
                     <h3 className="logo">Secret Admirer</h3>
                 </Link>
                 <ul className = "nav-links">
-                    <li><NavLink to="/findstar" className="navHeadings" activeClassName="current">Find My Star</NavLink></li>
-                    <li><NavLink to="/mystar" className="navHeadings" activeClassName="current">My Star</NavLink></li> 
-                    <li><NavLink to="/saved" className="navHeadings" activeClassName="current">Saved</NavLink></li> 
+                    <li><NavLink to="/saved" className="navHeadings" activeClassName="current">Saved</NavLink></li>
+                    {(this.state.username === "") ? (
+                        <li onClick={this.openModal} className="navHeadings">Log In to Account</li>
+                    ) :
+                        <li className="navHeadings">Welcome {" " + this.state.username + "!"} </li>
+                    }
```

Comment:
```
{(this.state.username === "") ? 
   <li onClick={this.openModal} className="navHeadings">Log In to Account</li> :
   <li className="navHeadings">Welcome {" " + this.state.username + "!"} </li>}
```
does this work?

Reply:
yes 
Nav renders with Log In to Account and it changes to welcome + username when user signs In

Reply:
No I removed () and rewrote used different styling. check if updated code works